src/pipeline.py

```diff
@@ -46,44 +43,41 @@
         "max_aspect_ratio": 2.5,
     },
 )
-caption_images_op = ComponentOp.from_registry(
-    name="caption_images",
+
+captions = images.apply(
+    "caption_images",
     arguments={
         "model_id": "Salesforce/blip-image-captioning-base",
         "batch_size": 8,
         "max_new_tokens": 50,
     },
-    number_of_accelerators=1,
-    accelerator_name="GPU",
 )
-segment_images_op = ComponentOp.from_registry(
-    name="segment_images",
+
+segmentations = captions.apply(
+    "segment_images",
     arguments={
         "model_id": "openmmlab/upernet-convnext-small",
         "batch_size": 8,
     },
-    number_of_accelerators=1,
-    accelerator_name="GPU",
 )
 
-# Add the component operations to your pipeline and define the dependencies
-pipeline.add_op(generate_prompts_op)
-pipeline.add_op(laion_retrieval_op, dependencies=generate_prompts_op)
-pipeline.add_op(download_images_op, dependencies=laion_retrieval_op)
-pipeline.add_op(caption_images_op, dependencies=download_images_op)
-pipeline.add_op(segment_images_op, dependencies=caption_images_op)
 
 # OPTIONAL: writing the dataset to the Hugging Face Hub
 HF_USER = None  # Insert your huggingface username here
 HF_TOKEN = None  # Insert your HuggingFace token here
 if HF_USER and HF_TOKEN:
-    write_to_hub_controlnet = ComponentOp(
-        component_dir="components/write_to_hub_controlnet",
+    segmentations.apply(
```

Comment:
```suggestion
    segmentations.write(
```
Didn't think of this before, but `apply` is indeed a superset of `write`, so will work for any write component as well.